analyzers/src/SonarAnalyzer.Common/Helpers/WildcardPatternMatcher.cs

```diff
@@ -0,0 +1,134 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Concurrent;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.Helpers;
+
+internal static class WildcardPatternMatcher
+{
+    public static bool IsMatch(string pattern, string input) =>
+        !(string.IsNullOrWhiteSpace(pattern) || string.IsNullOrWhiteSpace(input))
+        && WildcardPattern.Create(pattern).Match(input);
+
+    /// <summary>
+    /// Copied from https://github.com/SonarSource/sonar-plugin-api/blob/a9bd7ff48f0f77811ed909070030678c443c975a/sonar-plugin-api/src/main/java/org/sonar/api/utils/WildcardPattern.java.
+    /// </summary>
+    private sealed class WildcardPattern
+    {
+        private const string SpecialChars = "()[]^$.{}+|";
+        private static readonly ConcurrentDictionary<string, WildcardPattern> Cache = new();
+        private readonly Regex pattern;
+
+        private WildcardPattern(string pattern, string directorySeparator) =>
+            this.pattern = new Regex(ToRegexp(pattern, directorySeparator), RegexOptions.Compiled, RegexConstants.DefaultTimeout);
+
+        public bool Match(string value)
+        {
+            value = value.TrimStart('/');
+            value = value.TrimEnd('/');
+            try
+            {
+                return pattern.IsMatch(value);
+            }
+            catch (RegexMatchTimeoutException)
+            {
+                return false;
+            }
+        }
+
+        public static WildcardPattern Create(string pattern) =>
+            Create(pattern, Path.DirectorySeparatorChar.ToString());
```

Comment:
`Path.DirectorySeparatorChar` is platform specific that will never ever change during runtime. So there's no reason to pass it around and store, when it's already stored in `Path.DirectorySeparatorChar` itself. Just use it from there where it's needed.

Reply:
Fixed by #6919